### PR TITLE
Adapted to Puppet 5

### PR DESCRIPTION
**Pull request by fmaciask**
I made this module to request hiera lookup for puppet version 5. Also, I fix the way it creates chroot directory, as it can be requested to be recursive.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -6,18 +6,25 @@
...
+    creates => $chroot_base,
+    command => "/bin/mkdir -p ${chroot_base}",
+  } -> file { $chroot_base: #}
+#  file { $chroot_base:
     ensure => 'directory',
     owner  => 'root',
     group  => 'root',
     mode   => '0755',
   }
+create_resources('sftp_jail::user', lookup('sftp_jail::user', {merge => 'deep'}))
```
Comment (ekohl) [inline manifests/init.pp]: It's preferred create actual parameters and [set the merge behavior in hiera data](https://puppet.com/docs/puppet/5.4/hiera_merging.html#configure-merge-behavior-in-hiera-data).

File: manifests/init.pp [hunk 2]
```diff
@@ -6,18 +6,25 @@
...
 class sftp_jail (
   Stdlib::Absolutepath $chroot_base = '/chroot',
 ) inherits ::sftp_jail::params {
 
-  file { $chroot_base:
+# To make sure it creates recursively the directory
+  exec { "Create ${chroot_base}":
+    creates => $chroot_base,
+    command => "/bin/mkdir -p ${chroot_base}",
+  } -> file { $chroot_base: #}
```
Comment (ekohl) [inline manifests/init.pp]: Could you remove the commented code?

File: manifests/init.pp (hunk 1 above)
Comment (alexjfisher) [reply manifests/init.pp]: The parameter should probably be `users` (plural). I think iteration is preferred over `create_resources` too.

ie something a bit like...
```puppet
$users.each |String $key, Hash $attrs| {
  sftp_jail::user { $key:
    * => $attrs,
  }
}
```

**Comment by fmaciask** [comment]
Removed comments and resource definitions in init.pp;.

File: manifests/init.pp (hunk 2 above)
Comment (fmaciask) [reply manifests/init.pp]: Done in latest commit.

File: manifests/init.pp (hunk 1 above)
Comment (fmaciask) [reply manifests/init.pp]: Removed create_resources in the latest commit.

File: README.md [hunk 3]
```diff
@@ -53,6 +54,27 @@ Now add a user to your jail.
...
 
+#### Hiera
+
+```puppet
+---
+classes:
+  - ssh
+  - sftp_jail
+sftp_jail::chroot_base: '/var/test'
+sftp_jail::jail:
```
Comment (alexjfisher) [inline README.md]: In removing the create_resources, this doesn't do anything anymore?

File: README.md [hunk 4]
```diff
@@ -53,6 +54,27 @@ Now add a user to your jail.
 
 To share write access to `/incoming`, set the `group` parameter of `sftp_jail::jail` to a group that is common to both users, such as the `sftpusers` group that you might use to redirect users into a jail.
 
+#### Hiera
+
+```puppet
```
Comment (bastelfreak) [inline README.md]: the syntax is yaml, not puppet

**Comment by bastelfreak** [comment]
Hi @fmaciask, thanks for the PR. Can you take a look at the suggestions above? Please also review the used email address in the commits, it isn't associated with your github account.

**Comment by bastelfreak** [comment]
Ping @fmaciask, are you still interested in this code?

**Comment by bastelfreak** [comment]
Hi,
I'm going to close this PR due to inactivity. Please reopen and rebase it if you're still interested.

**Pull request closed without merging**